### ISSUE TITLE
Use an async I/O-based http server

### ISSUE THREAD

**Issue by ysimonson** [braidery/braid#11]
We're currently constructing heavy-weight threads for each request. Async I/O may act as a large performance win. This is blocked until there's a way to do this in a battle-tested web framework that works on rust stable.

**Comment by jasonwyatt** [comment]
Are we constructing the threads on each request, or do the requests use a thread pool? I would think the latter..

**Comment by ysimonson** [comment]
I'm not sure, whatever the iron default is. I'd guess the former.

**Comment by jasonwyatt** [comment]
IIRC, iron uses a thread pool.. I think I saw it was using 32 threads on my toy project.

**Comment by ysimonson** [comment]
Yup, you're right! https://www.reddit.com/r/rust/comments/2vzxjr/poor_http_performance_iron_framework/con7l0v/
